lib/vector-buffers/src/topology/channel/sender.rs

```diff
@@ -164,9 +173,15 @@ impl<T: Bufferable> BufferSender<T> {
     }
 
     /// Configures this sender to instrument the items passing through it.
-    pub fn with_instrumentation(&mut self, handle: BufferUsageHandle) {
+    pub fn with_usage_instrumentation(&mut self, handle: BufferUsageHandle) {
         self.instrumentation = Some(handle);
     }
+
+    /// Configures this sender to instrument the send duration.
+    pub fn with_send_duration_instrumentation(&mut self, stage: usize, span: &Span) {
+        let _enter = span.enter();
+        self.send_duration = Some(register(BufferSendDuration { stage }));
+    }
 }
```

Comment:
TIOLI: you _could_ make these return `Self` if you wanted, so that they were more builder-y. Doing so would allow you to clean up their usages above to avoid having the `mut` locals, etc.